cmd/controller/main.go

```diff
@@ -109,6 +111,7 @@ func main() {
 		log.Fatal(http.ListenAndServe(":"+port, mux))
 	}()
 
+	ctx = filteredinformerfactory.WithSelectors(ctx, v1beta1.ManagedByLabelKey)
```

Comment:
I'm a bit confused about how this works, so I'd like to check that my understanding is correct.

This `WithSelectors` modifies the `ctx` to make it possible to filter on the managed-by label, which we take advantage of later in taskrun/controller.go#55 to create an informer with that ability. Is that correct?

That's a bit... subtle... but I don't know of a better way myself either, and the benefit seems clear.

Is there any kind of test you could add that would demonstrate that the controller doesn't get informed of a Pod it's not managing? That would help ensure that the logic is correct, and guard against future regressions.

Reply:
Thanks for the review Jason!

> This WithSelectors modifies the ctx to make it possible to filter on the managed-by label, which we take advantage of later in taskrun/controller.go#55 to create an informer with that ability. Is that correct?

Yes, correct.
- First of all, we need to create a FilteredInformerFactory to setup the FilteredPodInformer, we can pass an label array to filter and cache resources by using multiple labels.
- Then in the taskrun controller logic, when getting pod resource from informer (taskrun/controller.go#55), the Tekton controller logic can choose ONE of label cached resource from informer store, like the managed-by label we use

> Is there any kind of test you could add that would demonstrate that the controller doesn't get informed of a Pod it's not managing? That would help ensure that the logic is correct, and guard against future regressions.

Actually, I tried to add an unit test to cover the fail case. but I found it is hard to me to do that. :(
First of all, it is a framework change, not a functional change
I tried to create the context but without selector by using the `knative.dev/pkg/reconciler/testing` original `SetupFakeContext` function. Yes, it will report an error:
```
=== RUN   TestFilteredPodInformer
    logger.go:130: 2021-01-29T11:02:28.254+0800	PANIC	fake/fake_filtered_factory.go:47	Unable to fetch labelkey from context.
--- FAIL: TestFilteredPodInformer (0.00s)
panic: Unable to fetch labelkey from context. [recovered]
	panic: Unable to fetch labelkey from context.
```
But it is a `panic`, the `SetupFakeContext` function cannot return an error so that I can check if it is the error of `Unable to fetch labelkey from context.` So we gave up and assume, if in future we miss the WithSelector in context when setting up the filetered pod informer, we will see the same error and all related unit test will fail too. Like the tests which I modified:
- pkg/reconciler/pipelinerun/metrics_test.go
- pkg/reconciler/taskrun/metrics_test.go
- test/v1alpha1/wait_test.go

Or please let me know if you have any good idea! Thanks! :)